web/src/app/schedules/temp-sched/TempSchedShiftsList.tsx

```diff
@@ -49,13 +65,20 @@ export default function TempSchedShiftsList({
         DateTime.fromISO(s.start, { zone }),
         DateTime.fromISO(s.end, { zone }),
       ),
+      isValid: !isAfter(start, s.start) && !isBefore(end, s.end),
     }))
 
     if (!shifts.length) return []
 
+    const scheduleDawn = DateTime.fromISO(start, { zone }).startOf('day')
+    const scheduleDusk = DateTime.fromISO(end, { zone }).endOf('day')
+
+    const firstShiftDawn = shifts[0].start.startOf('day')
+    const lastShiftDusk = shifts[shifts.length - 1].end.endOf('day')
```

Comment:
This won't give you the last shift, they are sorted by start time, not end time. We will need to reduce through the list to get the last end time.